src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -179,7 +179,23 @@ void QgsValueRelationWidgetWrapper::setValue( const QVariant &value )
   }
   else if ( mComboBox )
   {
-    mComboBox->setCurrentIndex( mComboBox->findData( value ) );
+    // findData fails to tell a 0 from a NULL
+    // See: "Value relation, value 0 = NULL" - https://issues.qgis.org/issues/19981
+    int idx = -1; // default to not found
+    for ( int i = 0; i < mComboBox->count(); i++ )
+    {
+      QVariant v( mComboBox->itemData( i ) );
+      if ( ( v.isNull() && !value.isNull() ) || ( !v.isNull() && value.isNull() ) )
```

Comment:
This could be simplified to

```
if ( v.isNull() != value.isNull() && v == value )
{
  idx = i;
  break;
}
```

I thought we had this method around in a common library already, this had to be introduced in so many places... But can't find it right now.

Reply:
I think it would make for a really good method next to those here:
https://github.com/qgis/QGIS/blob/57f8ff7b594ec30fcebff89263bdb46734e4b0b7/src/core/qgis.h#L443-L460

Reply:
@elpaso any feedback on the second comment?

Reply:
@m-kuhn oh, sorry, I missed that second one.

I can add that in another PR, do you have an idea for the signature?

Reply:
Like this I would say:

```
CORE_EXPORT bool qgsVariantEqual( const QVariant &lhs, const QVariant &rhs );
```